### PR TITLE
boot.loader.systemd-boot: add extraInstallCommands option

### PR DESCRIPTION
**Pull request by parthy**
###### Description of changes

GRUB received the ability to run custom scripts after the installation in #108242. A comparable mechanism is not available for systemd-boot. This patch is an attempt at adding that functionality.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by messemar** [comment]
@Lassulus do you know how can review this ?

File: nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix [hunk 1]
```diff
@@ -57,6 +57,11 @@ let
       --disallow-untyped-defs \
       $out
   '';
+
+  finalSystemdBootBuilder = pkgs.writeScript "install-systemd-boot.sh" (''
+    #!${pkgs.runtimeShell}
+    ${checkedSystemdBootBuilder} $@
```
Comment (Lassulus) [inline nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix:63]:
```suggestion
    ${checkedSystemdBootBuilder} "$@"
```

File: nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix [hunk 2]
```diff
@@ -57,6 +57,11 @@ let
...
+  '' + cfg.extraInstallCommands);
```
Comment (Lassulus) [inline nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix:64]:
```suggestion
  finalSystemdBootBuilder = pkgs.writeScript "install-systemd-boot.sh" ''
    #!${pkgs.runtimeShell}
    ${checkedSystemdBootBuilder} $@
    ${cfg.extraInstallCommands}
  '';
```

File: nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix [hunk 3]
```diff
@@ -99,6 +105,16 @@ in {
       '';
     };
 
+    extraInstallCommands = mkOption {
+      default = "";
+      example = "";
+      type = types.lines;
+      description = lib.mdDoc ''
```
Comment (Lassulus) [inline nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix:112]: maybe an example would be nice? just to know for what this could be used

Comment (parthy) [reply nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix:116]: I've added an example I was currently playing with. Maybe there is a better way to do that, but I've found that this mechanism that the GRUB module already has can be very helpful when dealing with custom boot entries.

**Review by Lassulus — APPROVED** [review]

**Pull request merged**